src/core.rs

```diff
@@ -371,6 +451,86 @@ pub fn find_words(line: &str) -> impl Iterator<Item = Word> {
     })
 }
 
+/// Split `line` into words using the Unicode line breaking algorithm.
+#[cfg(feature = "unicode-linebreak")]
+fn find_unicode_words(line: &str) -> impl Iterator<Item = Word> + '_ {
+    // Strip all ANSI escape sequences from `text`.
+    fn strip_ansi_escape_sequences(text: &str) -> String {
+        let mut result = String::with_capacity(text.len());
+
+        let mut chars = text.chars();
+        while let Some(ch) = chars.next() {
+            if skip_ansi_escape_sequence(ch, &mut chars) {
+                continue;
+            }
+            result.push(ch);
+        }
+
+        result
+    }
+
+    // Construct an iterator over (original index, stripped index)
+    // tuples. We find the Unicode linebreaks on a stripped string,
+    // but we need the original indices so we can form words based on
+    // the original string.
+    let mut last_stripped_idx = 0;
+    let mut char_indices = line.char_indices();
+    let mut idx_map = std::iter::from_fn(move || match char_indices.next() {
```

Comment:
This is all terribly complex and somewhat slow — I'll rework it to have a fast path for the case where we don't see any `CSI` characters in the string.

Reply:
Will do this in #333.